docs/core-concepts/index.md

```diff
@@ -0,0 +1,49 @@
+# Core Concepts
+
```

Comment:
Reminder to include "This document is a work in progress."(eg. https://tekton.dev/docs/concepts)